File: system-test/java/src/main/java/com/optivem/eshop/dsl/core/usecase/external/clock/ClockDsl.java
package com.optivem.eshop.dsl.core.usecase.external.clock;

import com.optivem.eshop.dsl.driver.port.external.clock.ClockDriver;
import com.optivem.eshop.dsl.core.usecase.external.clock.usecases.GetTime;
import com.optivem.eshop.dsl.core.usecase.external.clock.usecases.GoToClock;
import com.optivem.eshop.dsl.core.usecase.external.clock.usecases.ReturnsTime;
import com.optivem.eshop.dsl.common.Closer;
import com.optivem.eshop.dsl.core.shared.UseCaseContext;

public class ClockDsl implements AutoCloseable {
    protected final ClockDriver driver;
    protected final UseCaseContext context;

    public ClockDsl(ClockDriver driver, UseCaseContext context) {
        this.driver = driver;
        this.context = context;
    }

    @Override
    public void close() {
        Closer.close(driver);
    }

    public GoToClock goToClock() {
        return new GoToClock(driver, context);
    }

    public ReturnsTime returnsTime() {
        return new ReturnsTime(driver, context);
    }

    public GetTime getTime() {
        return new GetTime(driver, context);
    }
}

File: system-test/typescript/src/testkit/dsl/core/usecase/external/clock/ClockDsl.ts
import type { Result } from '../../../../../common/result.js';
import type { ClockDriver } from '../../../../../driver/port/external/clock/clock-driver.js';
import type { ClockErrorResponse } from '../../../../../driver/port/external/clock/dtos/ClockErrorResponse.js';
import type { UseCaseContext } from '../../../use-case-context.js';

export class ClockDsl {
  constructor(
    private readonly driver: ClockDriver,
    private readonly context: UseCaseContext,
  ) {}

  async goToClock(): Promise<Result<void, ClockErrorResponse>> {
    return this.driver.goToClock();
  }

  async close(): Promise<void> {
    await this.driver.close();
  }
}
